main.go

```diff
@@ -26,8 +31,8 @@ func main() {
 		g.Error(err, "reading input")
 	}
 
-	if err := proto.Unmarshal(data, g.Request); err != nil {
-		g.Error(err, "parsing input proto")
+	if err2 := proto.Unmarshal(data, g.Request); err2 != nil {
```

Comment:
I think the problem here was shadowing, but no need to declare an `err2`. I think you could just as well do:

```
if err = proto.Unmarshal(data, g.Request); err != nil {
```

Just don't use the declaration and assignment sign `:=` there but `=` instead to reuse the first error variable declared above.

Reply:
> I think the problem here was shadowing

Yes it was

I confess I didn't look enough if the first `err`s were used later in the function or not, will check more properly